dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -613,6 +618,21 @@ private void handleEntry2tsv(List<String> args) throws ApiException, IOException
     }
 
     /**
+     * this function will validate CWL file
+     * using this command: cwltool --non-strict --validate <file_path>
+     * @param cwlFilePath
+     */
+    private boolean validateCWL(String cwlFilePath) {
+        final String[] s = { "cwltool", "--non-strict", "--validate", cwlFilePath };
+        try {
+            io.cwl.avro.Utilities.executeCommand(Joiner.on(" ").join(Arrays.asList(s)), false,  Optional.absent(), Optional.absent());
+        } catch (Exception ex) {
+            return false;
```

Comment:
Don't completely catch all exceptions are validation errors (for example, if cwltool was installed incorrectly). 
There should be three cases
1) passing validation
2) failing validation 
3) throw exception on unknown cases